example/Example.js

```diff
@@ -46,11 +43,51 @@ class Example extends Component {
       });
     }
 
-    onInit(){
-      Countly.pushTokenType(Countly.messagingMode.DEVELOPMENT, "Channel Name", "Channel Description");
-      Countly.enableLogging();
-      Countly.init("https://trinisoft.count.ly", "f0b2ac6919f718a13821575db28c0e2971e05ec5");
+    onInit = async() => {
+      if(!await Countly.isInitialized()) {
+        /**
+         * Recommended settings for Countly initialisation
+         * 
+         * @function setLoggingEnabled enable countly internal debugging logs
+         * @function enableCrashReporting enable crash reporting to report crash to Countly
+         * @function setRequiresConsent set consent should be required
+         * @function setViewTracking enable automatic view tracking
+         * 
+         */
+        Countly.setLoggingEnabled(true);
+        Countly.enableCrashReporting();
+        Countly.setRequiresConsent(true);
+        Countly.setViewTracking(true);
+
+        /**
+         * Optional settings for Counlty initialisation
+         * 
+         * @function enableParameterTamperingProtection set the optional salt to be used for calculating the checksum of requested data which will be sent with each request, using the &checksum field
+         * @function pinnedCertificates it will ensure that connection is made with one of the public keys specified
+         * @function setHttpPostForced Set HTTP POST is used in all cases
+         * @function enableApm  enable record app start time 
+         * @function enableAttribution enable to measure your marketing campaign performance by attributing installs from specific campaigns.
+         */
+        Countly.enableParameterTamperingProtection("salt");
+        // Countly.pinnedCertificates("count.ly.cer");
+        // Countly.setHttpPostForced(false);
+        Countly.enableApm();
+        Countly.enableAttribution();
+
+        /**
+         * @function setEventSendThreshold Set event threshold value, Events get grouped together and are sent either every minute or after the unsent event count reaches a threshold. By default it is 10
+         */
+        Countly.setEventSendThreshold("3");
+
+        /**
+         * @function pushTokenType set Push notification messaging mode and callbacks for push notifications
+         */
+        Countly.pushTokenType(Countly.messagingMode.DEVELOPMENT, "Channel Name", "Channel Description");
+
+        Countly.init("https://try.count.ly", "COUNTLY_APP_KEY");
```

Comment:
missing calls:
* that give consent (in this case we can just "giveAll")
* registers for push notification callbacks (registerForNotification)
* enables push notificaions (askForNotificationPermission)